dbms/src/DataStreams/WindowBlockInputStream.cpp

```diff
@@ -678,7 +794,7 @@ RowNumber WindowTransformAction::getPreviousRowNumber(const RowNumber & row_num)
     }
 
     --prev_row_num.block;
-    assert(prev_row_num.block - first_block_number < window_blocks.size());
+    assert(static_cast<Int64>(prev_row_num.block) - static_cast<Int64>(first_block_number) < static_cast<Int64>(window_blocks.size()));
```

Comment:
maybe `assertI(prev_row_num.block < window_blocks.size() + first_block_number)` is better?

Reply:
> assertI(prev_row_num.block < window_blocks.size() + first_block_number)

done